### PR TITLE
internal/website/content/howto/pubsub: use godoc examples

### LINKED ISSUE #1827: internal/website: how-to example code can drift from godoc examples
### Is your feature request related to a problem? Please describe.

The how-to guides under [internal/website/content/howto](https://github.com/google/go-cloud/tree/master/internal/website/content/howto) often duplicate example code from the godoc reference, but sometimes with slight differences.

Long-term, this can cause maintenance burden as improvements in one rendering likely should carry over into the other rendering. Further, the Hugo-pasted example code does not ever get compiled, so it is more likely that the code may not compile correctly.

### Describe the solution you'd like

The Hugo site should somehow reference/copy example code from the godoc examples. The hackiest way of implementing this would be a presubmit check that fails if skew occurs, but ideally the Hugo site could reference examples from godoc, preferably with a way of doing some minimal rewrites (e.g. `log.Fatal(err)` -> `return err` and removing blank identifier assignments).

### Describe alternatives you've considered

I've considered making the how-to guides in Hugo be the canonical source of truth, as we definitely need them in the how-to guides. However, I rejected this because:

-  many people already consume documentation in godoc.org directly
-  examples in godoc get compiled and optionally run, improving confidence in their correctness
-  making the user navigate back and forth just to show examples seems unreasonable

### PR DESCRIPTION
**Pull request by zombiezen**
Required some restructuring to existing Pub/Sub examples, but should make them more closely associated with the API they are exemplifying.

Also mark examples.json and replay YAML files as generated, see https://thoughtbot.com/blog/github-diff-supression.

Fixes #1827

### CODE REVIEW

File: .gitattributes [hunk 1]
```diff
@@ -0,0 +1,2 @@
+/internal/website/data/examples.json  linguist-generated=true
+**/testdata/**/*.yaml  linguist-generated=true
```
Comment (vangent) [inline .gitattributes]: Please add `**/testdata/**/*.replay` as well. Some of them are binary, so are probably collapsed by default, but others are text (recorded HTTP requests/responses).

File: internal/website/content/howto/pubsub/subscribe.md [hunk 2]
```diff
@@ -555,19 +225,7 @@ You will also need to pass an acknowledgement deadline: once a message is
...
-    "gocloud.dev/pubsub/mempubsub"
-)
-
-// ...
-
-topic := mempubsub.NewTopic()
-defer topic.Shutdown(ctx)
-subscription := mempubsub.NewSubscription(topic, 1 * time.Minute)
-```
+{{< goexample "gocloud.dev/pubsub/mempubsub.ExampleNewSubscription" >}}
```
Comment (vangent) [inline internal/website/content/howto/pubsub/subscribe.md]: Question: if you typo this and `ExampleNewSubscription` does not exist, will Travis catch that?

File: .gitattributes (hunk 1 above)
Comment (zombiezen) [reply .gitattributes]: Done.

File: internal/website/content/howto/pubsub/subscribe.md (hunk 2 above)
Comment (zombiezen) [reply internal/website/content/howto/pubsub/subscribe.md]: Good point. Added logic to the shortcode to make it do exactly that.

**Pull request merged**